sim/dex.ts

```diff
@@ -363,14 +363,27 @@ export class ModdedDex {
 		}
 		if (!this.data.Pokedex.hasOwnProperty(id)) {
 			let aliasTo = '';
-			if (id.startsWith('mega') && this.data.Pokedex[id.slice(4) + 'mega']) {
-				aliasTo = id.slice(4) + 'mega';
-			} else if (id.startsWith('m') && this.data.Pokedex[id.slice(1) + 'mega']) {
-				aliasTo = id.slice(1) + 'mega';
-			} else if (id.startsWith('primal') && this.data.Pokedex[id.slice(6) + 'primal']) {
-				aliasTo = id.slice(6) + 'primal';
-			} else if (id.startsWith('p') && this.data.Pokedex[id.slice(1) + 'primal']) {
-				aliasTo = id.slice(1) + 'primal';
+			const formeNames: {[k: string]: string[]} = {
+				alola: ['a', 'alola', 'alolan'],
+				galar: ['g', 'galar', 'galarian'],
+				gmax: ['gigantamax', 'gmax'],
+				mega: ['m', 'mega'],
+				primal: ['p', 'primal'],
+			};
+			for (const forme in formeNames) {
```

Comment:
I don't like this algorithm. It fails on Mega Altaria, since that will be converted to `megaaltarialola` and then fail to show up.

Reply:
Oh, wait, you don't have early breaks. You have the opposite problem: `alolangolem` will be converted to `alolangolemega` and then fail.

Reply:
Oh, wait, you are testing them all separately.

Reply:
Yeah, this is fine. Non-ideal, but it won't have any weird misbehavior.

Reply:
So this is good to go right